src/internal/_xdropRepeatsWith.js

```diff
@@ -0,0 +1,22 @@
+var _curry2 = require('./_curry2');
+var last = require('../last');
+
+
+module.exports = (function() {
+  function XDropRepeatsWith(pred, xf) {
+    this.xf = xf;
+    this.pred = pred;
+  }
+  XDropRepeatsWith.prototype['@@transducer/init'] = function() {
+    return this.xf.init();
+  };
+  XDropRepeatsWith.prototype['@@transducer/result'] = function(result) {
+    return this.xf['@@transducer/result'](result);
+  };
+  XDropRepeatsWith.prototype['@@transducer/step'] = function(result, input) {
+    var sameAsLast = result.length !== 0 && this.pred(last(result), input);
```

Comment:
You should not make any assumptions of the accumulator type when implementing transducers.  The `result` here could be anything from an array, to a stream, channel, observable, etc.  The only thing you should do with `result` is optionally pass it through `@@transducer/step` and return.